src/lib/functions/netlify-function.js

```diff
@@ -39,25 +45,29 @@ class NetlifyFunction {
   // - `srcFilesDiff`: Files that were added and removed since the last time
   //    the function was built.
   async build({ cache }) {
-    const buildFunction = await this.runtime.getBuildFunction({
-      config: this.config,
-      errorExit: this.errorExit,
-      func: this,
-      functionsDirectory: this.functionsDirectory,
-      projectRoot: this.projectRoot,
-    })
-
-    this.buildQueue = buildFunction({ cache })
-
     try {
-      const { srcFiles, ...buildData } = await this.buildQueue
-      const srcFilesSet = new Set(srcFiles)
-      const srcFilesDiff = this.getSrcFilesDiff(srcFilesSet)
+      return await this.mutex.runExclusive(async () => {
```

Comment:
This is just another way of doing the same thing. Happy to revert it

Reply:
Reverted in https://github.com/netlify/cli/pull/2930/commits/14dfe1d7e029da6d48aa9fd6334031638686e487 for clarity